### ISSUE TITLE
Please do not change future options

### ISSUE THREAD

**Issue by HenrikBengtsson** [martonkolossvary/parseRPDR#1]
Hello. Please do _not_ set future options from within your package, as in:

https://github.com/martonkolossvary/parseRPDR/blob/aec45d769647a5f79e9e8aceffda5151e4f9811f/R/ZZZ.R#L1-L4

R options are meant for end-users, not developers. If you need to tweak the settings, please do so temporarily, as explained in https://future.futureverse.org/articles/future-7-for-package-developers.html#avoid-changing-the-future-options.

**Comment by CAndrewBasham** [comment]
Hi sorry in not clear how I did that. Or what that means...

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Henrik Bengtsson ***@***.***>
Sent: Thursday, February 29, 2024 6:47:32 PM
To: martonkolossvary/parseRPDR ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [martonkolossvary/parseRPDR] Please do not change future options (Issue #1)


        External Email - Use Caution

Hello. Please do not set future options from within your package, as in:

https://github.com/martonkolossvary/parseRPDR/blob/aec45d769647a5f79e9e8aceffda5151e4f9811f/R/ZZZ.R#L1-L4<https://secure-web.cisco.com/1PLflBLxX6gD5vvWsbtwgou71veoB36W_9Nr8d-K9gW6yPIPTtdAD_SSFx073hGmfKNb4woBqitshUF2Oyi4t7soDFB4aEMFnv1cMre_7KoyhvXJyntfnsKDPlokQ5U5rn8XtKKZXDcyVYZZBp8amhnbY0cV5fA0rnzszaO_pgUo9D4sqUFz2ujzOWirqxGj8YCAHuocw_N5vCC3v8Fial442WDsTJJsl04AcgO-SKq4BNJAKCADINadviknZ5-CUAT6Uc72prUOHF1atQ1OMpBjmOdHgoibPQgIHhgLkBxrBkhKQd_F9d3LtTXI_DevP/https%3A%2F%2Fgithub.com%2Fmartonkolossvary%2FparseRPDR%2Fblob%2Faec45d769647a5f79e9e8aceffda5151e4f9811f%2FR%2FZZZ.R%23L1-L4>

R options are meant for end-users, not developers. If you need to tweak the settings, please do so temporarily, as explained in https://future.futureverse.org/articles/future-7-for-package-developers.html#avoid-changing-the-future-options<https://secure-web.cisco.com/1o4uT0tEW6-JFAjk3GhWvWrmA89NZOZdy6CgVoeiQZ4CPSqQXX-3RUpChY0GGXSU-dHGWppQt61Ez53PT6nvRPhIQXTXF_KIE4CwWcEdMSLoWgGfTUzQEFU_0Dnmis7mr1lTwk3oYvD115V9JECNcAq0ccDVUMlfEjRobD4k_fVvydJ9qX2VQJBx0GX9dhb_ZksDSmmqNiX151zLQd77GzW8eRe3J9NrAFFBIzTlC86W80GVY9H4iJfRWD2bbFKg6ucp0Hxb1q2OJ6zscTTw3WZta70e2R28DN3CgKJCUkeXP0Eoc7k2xVqN1Je8DuPAX/https%3A%2F%2Ffuture.futureverse.org%2Farticles%2Ffuture-7-for-package-developers.html%23avoid-changing-the-future-options>.

—
Reply to this email directly, view it on GitHub<https://secure-web.cisco.com/1dClp15ah_L_JyKQw0FM8WG41uIe-9zs2Au5EOCCNKOeaq3iRgszyuCDjKSpL4P7YcI1Krq_VsD1kcBj2YoK0aHT5CT03WimarQABkfVkUBogomnSJfu7ib1ZBrpRDiaT7td1u1lGOTBnwvj5HaWpn8iS7Do2kkpaAU1-V_X-t3d6BcbP-2mYUMw9v6WMtAKsSH4C7dUHrv6h_2RFwHLT2iemy_B-sp3zB68rfOhtQEGTBim_huKorqTYEhoSyon1Btg0tbSYx5raFoaJZWt0EzLRg2iDKXSsdfi9scJIpmG7QGpdQU7G70PCV43eazZS/https%3A%2F%2Fgithub.com%2Fmartonkolossvary%2FparseRPDR%2Fissues%2F1>, or unsubscribe<https://secure-web.cisco.com/1QMFy6ncCel6NBcYlGZ_0Tt8I8U11s2NtdF18BOx3rFRVnI7WQg7XOHKIxjtYvkPyGWznr5QlDPStmuhgydJbA7M8Giy-d757oCnppTrDCH_JqwapbNnLpaBKv6UWNws-bR2m8WZR5oG0pTXRBln8669I13X5EVIwEoJbcV3yZy2GMOTzwSjCosccHUYt4KLgwJSppyW6xbcTG8_2A6gFbDi0n6Q7Gz5MhQ6W1peYjsCjPKD8Vk0RwG6qWgSuylayW-WbFmc4PbjNsD5xRrxwAEWNdcn9DiDfyU6qYfdEtDW0k8_-f58fgaYnK3mpwjv3/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FBEE5P3APCTT24IS65S4VEWLYV66ZJAVCNFSM6AAAAABEA2VLZWVHI2DSMVQWIX3LMV43ASLTON2WKOZSGE3DEMRYGE2TCNY>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
The information in this e-mail is intended only for the person to whom it is addressed.  If you believe this e-mail was sent to you in error and the e-mail contains patient information, please contact the Mass General Brigham Compliance HelpLine at https://www.massgeneralbrigham.org/complianceline <https://www.massgeneralbrigham.org/complianceline> .
Please note that this e-mail is not secure (encrypted).  If you do not wish to continue communication over unencrypted e-mail, please notify the sender of this message immediately.  Continuing to send or respond to e-mail after receiving this message means you understand and accept this risk and wish to continue to communicate over unencrypted e-mail.

**Comment by HenrikBengtsson** [comment]
Just checking back in on this one. I see you submitted on parseRPDR 1.1.1 on 2024-04-03 to CRAN, after I filed this issue.

**Comment by martonkolossvary** [comment]
Dear Henrik,

Thank you so much for pointing out this implementation of future. I
acknowledge that hard coding the parallelization is not optimal,
furthermore, it is not advised by the developers. However parseRPDR is a R
package designed to be used by medical professionals who do not know how to
code to be able to parse outputs of the Harvard patient data registry into
csv formats. Therefore 99.9% of the users do not know how to set the
backend for parallelization and do not wish to change it but want to have a
tool that works out of the box without needing any technical knowledge.
Therefore, I would like to keep the current implementation to support the
ease of use of the software package.

Thank you for your kind consideration.

Marton Kolossvary

**Comment by HenrikBengtsson** [comment]
Hi. 

Please see what I wrote and the link I refer to in my top comment. I'm asking you to not set this when the package is attached. You can still set it _temporarily_ inside the functions where you need it. <https://future.futureverse.org/articles/future-7-for-package-developers.html#avoid-changing-the-future-options> has almost cut'n'pastable instructions for how to do this.

Also, there are no future R options `future.globals` and `future.chunk.size`, so those you can drop.

**Comment by HenrikBengtsson** [comment]
I and lots of people using futureverse would appreciate it if you fixed this. As I said, the fix is possible.

**Comment by martonkolossvary** [comment]
To the best of my knowledge I have in the latest release on cran v1.1.2

**Comment by HenrikBengtsson** [comment]
Oh, I see. Great. Thanks. I didn't see any commits or updates here in this repository, which still has the old code - I guess they still need to be pushed here.

**Comment by martonkolossvary** [comment]
Yes, it takes some time until the package is also pushed to github not only
cran
